### PR TITLE
Decompile scrcmd_20

### PR DESCRIPTION
**Pull request by cbt6**

### CODE REVIEW

File: src/scrcmd_20.c [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+    u8 s = 0;
+    sub_02030AA4(r0, 0xa, r5 * 4 + r4, &s);
+    if (r4 == 3) {
+        u32 r7;
+        if (r5 == 0) {
+            r7 = 0x66;
+        } else {
+            r7 = 0x68;
+        }
+        sub_02031108(
```
Comment (red031000) [inline src/scrcmd_20.c:82]: please don't split params over lines like this

File: src/scrcmd_20.c [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+            sub_0205C268(sub_0205C11C(r5))
+        );
+        break;
+    case 3:
+        sub_0204F878(ctx->fsys->savedata, sp04, r5);
+        break;
+    case 4:
+        u32 challengeType;
+        if (r5 == 0) {
+            challengeType = 0;
```
Comment (red031000) [inline src/scrcmd_20.c:139]: use enum or defines

File: src/scrcmd_20.c [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+    default:
+        GF_ASSERT(FALSE);
+        *resultPtr = 0;
+    }
+    return FALSE;
+}
+
+static BOOL BattleHall_DoesPartyContainEligibleMons(s32 numRequiredMons, SAVEDATA *savedata) {
+    u8 i, numEligibleMons;
+    u16 species, level, forme;
```
Comment (red031000) [inline src/scrcmd_20.c:156]: I personally would prefer these to be each on their own separate lines

File: src/scrcmd_20.c [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+    r4->playerSpecies = playerSpecies;
+    r4->resultPtr = resultPtr;
+    sub_02091574((FieldSystem *)r4);
+    TaskManager_Call(taskManager, sub_0204F228, r4);
+}
+
+static BOOL sub_0204F228(TaskManager *taskManager) {
+    UnkStruct_0204F1E4 *r4 = TaskManager_GetEnv(taskManager);
+    switch (r4->state) {
+        case 0: {
```
Comment (red031000) [inline src/scrcmd_20.c:221]: please don't put braces around cases if not required

File: src/scrcmd_20.c [hunk 5]
```diff
@@ -0,0 +1,362 @@
...
+    partyMenu->unk_37 = 30;
+    partyMenu->unk_36_0 = 1;
+    partyMenu->unk_36_4 = 1;
+    partyMenu->fsys_unk_10C = &fsys->unk_10C;
+    if (a0->challengeType == 1) { // doubles
+        partyMenu->unk_36_0 = 2;
+        partyMenu->unk_36_4 = 2;
+    }
+    FieldSys_LaunchApplication(fsys, &_0210159C, partyMenu);
+    *a0->unk08 = partyMenu;
```
Comment (red031000) [inline src/scrcmd_20.c:306]: please put brackets around deferences like this to make it clear what is being dereferenced

**Review by red031000 — CHANGES_REQUESTED** [review]

File: include/unk_0205BFF0.h [hunk 6]
```diff
@@ -1,6 +1,9 @@
 #ifndef POKEHEARTGOLD_UNK_0205BFF0_H
 #define POKEHEARTGOLD_UNK_0205BFF0_H
 
+int sub_0205BFF0(u8, u8);
+int sub_0205C048(u8, u8);
+int sub_0205C11C(u8);
 u32 sub_0205C144(u8 a0);
 
 #endif //POKEHEARTGOLD_UNK_0205BFF0_H
```
Comment (adrienntindall) [inline include/unk_0205BFF0.h:9]: new line at end of file

File: include/unk_02088288.h [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+#include "scrcmd.h"
+
+extern struct OVY_MGR_TEMPLATE _02103A1C;
+
+u32 sub_02088288(SAVEDATA*);
+BOOL sub_0208828C(SAVEDATA*);
+void sub_02089D40(UnkStruct_0204F448*, const u8*);
+void sub_0208AD34(UnkStruct_0204F448*, PLAYERPROFILE*);
+
+#endif //POKEHEARTGOLD_UNK_02088288_H
```
Comment (adrienntindall) [inline include/unk_02088288.h:13]: new line at end of file

File: src/scrcmd_20.c [hunk 8]
```diff
@@ -0,0 +1,362 @@
...
+                && level >= 30
+                && IsPokemonBannedFromBattleFrontier(species, forme) != TRUE) {
+            numEligibleMons++;
+            arr[i] = species;
+        }
+    }
+    if (numRequiredMons == 2) {
+        for (u8 j = 0; j < partyCount; j++) {
+            for (i = 0; i < partyCount; i++) {
+                if (j != i
```
Comment (adrienntindall) [inline src/scrcmd_20.c:181]: these probably don't need to be on separate lines

File: src/scrcmd_20.c (hunk 5 above)
Comment (adrienntindall) [reply src/scrcmd_20.c:306]: also to add on to this you can prolly document unk08 in UnkStruct_0204F284 to be 'PartyMenuAppData *partyMenu' unless something later down the line contradicts that

File: src/scrcmd_20.c [hunk 9]
```diff
@@ -0,0 +1,362 @@
...
+        return 3;
+    }
+    UnkStruct_0204F448 *r0 = *a0->unk08;
+    a0->unk05 = r0->unk14;
+    FreeToHeap(r0);
+    *a0->unk08 = NULL;
+    return 0;
+}
+
+
```
Comment (adrienntindall) [inline src/scrcmd_20.c:362]: two newlines here at end of file, delete one

**Review by adrienntindall — CHANGES_REQUESTED** [review]

File: include/unk_02088288.h [hunk 10]
```diff
@@ -0,0 +1,13 @@
+#ifndef POKEHEARTGOLD_UNK_02088288_H
+#define POKEHEARTGOLD_UNK_02088288_H
+
+#include "scrcmd.h"
```
[the remark this answers is not in the document]
Comment (red031000) [reply include/unk_02088288.h:4]: like does the entirety of scrcmd.h need to be included? or just the constituent files?

[the remark this answers is not in the document]
Comment (cbt6) [reply include/unk_02088288.h:4]: The point is to include the definition of `UnkStruct_0204F448`.

[the remark this answers is not in the document]
Comment (red031000) [reply include/unk_02088288.h:4]: it's probably more appropriate to put that struct here and then include this header in scrcmd then

**Review by red031000 — APPROVED** [review]

**Review by adrienntindall — APPROVED** [review]

**Pull request merged**